packages/core/src/shared/errors.ts

```diff
@@ -760,7 +760,11 @@ export class PermissionsError extends ToolkitError {
 }
 
 export function isNetworkError(err?: unknown): err is Error & { code: string } {
-    if (isVSCodeProxyError(err)) {
+    if (!(err instanceof Error)) {
+        return false
+    }
+
+    if (isVSCodeProxyError(err) || isTimeoutError(err)) {
```

Comment:
is `isTimeoutError` always a network error? if we want to keep the semantics of `isNetworkError` , perhaps the caller should check `isNetworkError || isTimeoutError` instead of embedding it here

Reply:
In the aws-sdk pr linked above `TimeoutError` represents: `"ECONNRESET", "EPIPE", "ETIMEDOUT"`. Which all look to be network errors

Reply:
Seems likely that `err.name === 'TimeoutError'` could have false positives. Can we (also) check for `"ECONNRESET", "EPIPE", "ETIMEDOUT"` somehow? Even `message.includes()` would be better than nothing, to avoid false positives.

`isNetworkError` is an interface that claims to do something very specific. If it silently lies about certain errors, that can cause hidden bugs.

Reply:
Looked at the aws-sdk code and the `E...` errors are part of the `code` field of the error. So we can actually just add the missing error codes to our existing list and that should indirectly handle the TimeoutError.